# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/moz.build
# Commit: ec6a64b639a8
# Full Hash: ec6a64b639a82e957376c09a71aef7ce62b3a5aa
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-12-03 09:47:26
# Regressor Bug: 1657404
# File Overlap Count: 1
# Description:
#   Bug 1657404 - Implement a strongly-typed, service-dependent gamepad handle r=handyman,aklotz
#   
#   Currently, the gamepad code uses a uint32_t as a handle and does some trickery
#   with it by trying to create a unique ID and adding an offset to it for VR code.
#   
# ==============================================================================

diff -r 6792f28b99bd -r ec6a64b639a8 dom/gamepad/moz.build
--- a/dom/gamepad/moz.build	Wed Dec 02 22:36:35 2020 +0000
+++ b/dom/gamepad/moz.build	Wed Dec 02 23:06:05 2020 +0000
@@ -16,6 +16,7 @@
 EXPORTS.mozilla.dom += [
     "Gamepad.h",
     "GamepadButton.h",
+    "GamepadHandle.h",
     "GamepadHapticActuator.h",
     "GamepadLightIndicator.h",
     "GamepadManager.h",
@@ -30,7 +31,6 @@
     "ipc/GamepadEventChannelChild.h",
     "ipc/GamepadEventChannelParent.h",
     "ipc/GamepadMessageUtils.h",
-    "ipc/GamepadServiceType.h",
     "ipc/GamepadTestChannelChild.h",
     "ipc/GamepadTestChannelParent.h",
 ]
@@ -38,6 +38,7 @@
 UNIFIED_SOURCES = [
     "Gamepad.cpp",
     "GamepadButton.cpp",
+    "GamepadHandle.cpp",
     "GamepadHapticActuator.cpp",
     "GamepadLightIndicator.cpp",
     "GamepadManager.cpp",